src/lib/realmdb/RealmDB.js

```diff
@@ -91,10 +100,9 @@ class RealmDB implements DB, ProfileDB {
    */
   async init(db: ThreadDB) {
     try {
-      const { privateKey, Feed } = db
+      const { privateKey, wallet, Feed } = db
```

Comment:
why are we passing wallet? this is dangerous because wallet changes when switching networks. ThreadDB doesnt

Reply:
will check

Reply:
we do not create new wallet instance on network switch, just reinitialise it:

```
        await goodWallet.setIsPollEvents(false) //stop watching prev chain events
        await goodWallet.init({ network: contractsNetwork }) //reinit wallet
```

Reply:
that's why it is dangerous, if in a middle of some IO operation we change the wallet data and once that IO operation resumes it reads data from the wallet but now this data is of the new chain and not the chain the IO operation was for.

looking at nativetxssource it doesnt look like anything dangerous is used